theano/gpuarray/dnn.py

```diff
@@ -1382,69 +1427,75 @@ def make_node(self, dy, sm):
-                                 AbstractConv2d_gradInputs))):
-        return None
-
-    inp1 = node.inputs[0]
-    inp2 = node.inputs[1]
+@register_opt2([AbstractConv2d, AbstractConv2d_gradWeights,
+                AbstractConv2d_gradInputs], 'fast_compile')
+def local_abstractconv_cudnn_graph(op, context_name, inputs, outputs):
+    if (not isinstance(op, (AbstractConv2d,
+                            AbstractConv2d_gradWeights,
+                            AbstractConv2d_gradInputs))):
+        return
 
-    if not isinstance(inp1.type, GpuArrayType):
-        return None
+    inp1 = inputs[0]
+    inp2 = inputs[1]
 
     if not dnn_available(inp1.type.context_name):
         raise_no_cudnn()
 
-    if node.op.filter_flip:
+    if op.filter_flip:
         conv_mode = 'conv'
     else:
         conv_mode = 'cross'
 
-    if isinstance(node.op, AbstractConv2d):
+    if isinstance(op, AbstractConv2d):
         rval = dnn_conv(inp1, inp2,
-                        border_mode=node.op.border_mode,
-                        subsample=node.op.subsample,
+                        border_mode=op.border_mode,
+                        subsample=op.subsample,
                         direction_hint='forward!',
                         conv_mode=conv_mode)
-    if isinstance(node.op, AbstractConv2d_gradWeights):
+    elif isinstance(op, AbstractConv2d_gradWeights):
         shape = (inp2.shape[1], inp1.shape[1],
-                 node.inputs[2][0], node.inputs[2][1])
+                 inputs[2][0], inputs[2][1])
         rval = dnn_gradweight(inp1, inp2, shape,
-                              border_mode=node.op.border_mode,
-                              subsample=node.op.subsample,
+                              border_mode=op.border_mode,
+                              subsample=op.subsample,
                               conv_mode=conv_mode)
-    if isinstance(node.op, AbstractConv2d_gradInputs):
+    elif isinstance(op, AbstractConv2d_gradInputs):
         shape = (inp2.shape[0], inp1.shape[1],
-                 node.inputs[2][0], node.inputs[2][1])
+                 inputs[2][0], inputs[2][1])
         rval = dnn_gradinput(inp1, inp2, shape,
-                             border_mode=node.op.border_mode,
-                             subsample=node.op.subsample,
+                             border_mode=op.border_mode,
+                             subsample=op.subsample,
                              conv_mode=conv_mode)
     return [rval]
 
-conv_groupopt.register('local_abstractconv_cudnn',
+
+@local_optimizer([AbstractConv2d, AbstractConv2d_gradWeights,
+                  AbstractConv2d_gradInputs])
+def local_abstractconv_cudnn(node):
+    ctx = infer_context_name(*node.inputs)
+    if not isinstance(node.inputs[0].type, GpuArrayType):
+        return
+    return local_abstractconv_cudnn_graph(node.op, ctx, node.inputs, node.outputs)
+
+conv_groupopt.register('local_abstractconv_cudnn_graph',
```

Comment:
Since the method calls the `local_abstractconv_cudnn_graph` and the optimization is being done as a part of GraphToGpu, i thought of tagging it like this. Should i revert ?